test/legacy_test/test_sinc.py

```diff
@@ -0,0 +1,205 @@
+                            name='x', shape=shape, dtype=dtype
+                        )
+                        res = paddle.sinc(x)
+                        static_result = exe.run(
+                            feed={'x': x_data}, fetch_list=[res]
+                        )[0]
+                        out_expected = np_sinc(x_data)
+                    np.testing.assert_allclose(
+                        static_result, out_expected, rtol=1e-6, atol=1e-6
+                    )
+
+        for place in self.place:
+            run_dygraph(place)
+            run_static(place)
+
+    def test_zero(self):
+        def run_dygraph(place):
+            paddle.disable_static(place)
+            if core.is_compiled_with_cuda():
+                support_dtypes = self.cuda_support_dtypes
+            else:
+                support_dtypes = self.cpu_support_dtypes
+
+            for dtype in support_dtypes:
+                for shape in self.shapes:
+                    x_data = np.random.rand(*shape).astype(dtype)
+                    mask = (
+                        (np.random.rand(*shape) > 0.5)
+                        .astype('int')
+                        .astype(dtype)
+                    )
+                    x_data = x_data * mask
+                    x = paddle.to_tensor(x_data)
+                    out = paddle.sinc(x)
+                    out_expected = np_sinc(x_data)
+                    np.testing.assert_allclose(
+                        out.numpy(), out_expected, rtol=1e-6, atol=1e-6
+                    )
+
+        for place in self.place:
+            run_dygraph(place)
+
+    def test_input_type_error(self):
+        with self.assertRaises(TypeError):
+            x = np.random.rand(6).astype('float32')
+            x = paddle.sinc(x)
+
+    def test_input_dype_error(self):
+        paddle.enable_static()
+        place = paddle.CPUPlace()
+        with self.assertRaises(TypeError):
+            x_data = np.random.rand(6).astype('int32')
+            startup_program = paddle.static.Program()
+            main_program = paddle.static.Program()
+            exe = base.Executor(place)
+            with paddle.static.program_guard(main_program, startup_program):
+                x = paddle.static.data(name='x', shape=[6], dtype='int32')
+                res = paddle.sinc(x)
+                static_result = exe.run(feed={'x': x_data}, fetch_list=[res])[0]
+
+        with self.assertRaises(TypeError):
+            x_data = np.random.rand(6).astype('int64')
+            startup_program = paddle.static.Program()
+            main_program = paddle.static.Program()
+            exe = base.Executor(place)
+            with paddle.static.program_guard(main_program, startup_program):
+                x = paddle.static.data(name='x', shape=[6], dtype='int64')
+                res = paddle.sinc(x)
+                static_result = exe.run(feed={'x': x_data}, fetch_list=[res])[0]
+
+    def test_inplace(self):
```

Comment:
the type error of inplace API also should be test.

Reply:
added